src/fu-engine.c

```diff
@@ -1054,6 +1054,110 @@ fu_engine_check_hardware_requirement (FuEngine *self, AsApp *app, GError **error
 }
 #endif
 
+#if AS_CHECK_VERSION(0,7,8)
+static gchar *
+fu_engine_get_pkgconfig_version_fn (const gchar *pkgconfig_fn, GError **error)
+{
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) lines = NULL;
+
+	/* load file */
+	if (!g_file_get_contents (pkgconfig_fn, &data, NULL, error))
+		return NULL;
+
+	/* look for version */
+	lines = g_strsplit (data, "\n", -1);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		if (g_str_has_prefix (lines[i], "Version: "))
+			return g_strdup (lines[i] + 9);
+	}
+
+	/* nothing suitable found! */
+	g_set_error (error,
+		     FWUPD_ERROR,
+		     FWUPD_ERROR_INVALID_FILE,
+		     "Invalid pkg-config file found: %s",
+		     pkgconfig_fn);
+	return FALSE;
+}
+
+static gchar *
+fu_engine_get_pkgconfig_version (const gchar *pkgconfig_name, GError **error)
+{
+	g_autofree gchar *fn = g_strdup_printf ("%s.pc", pkgconfig_name);
+	const gchar *paths[] = { LIBDIR, DATADIR, "/usr/share", "/usr/lib64", NULL };
+	for (guint i = 0; paths[i] != NULL; i++) {
+		g_autofree gchar *path = NULL;
+		path = g_build_filename (paths[i], "pkgconfig", fn, NULL);
+		if (g_file_test (path, G_FILE_TEST_EXISTS))
+			return fu_engine_get_pkgconfig_version_fn (path, error);
```

Comment:
Considering how fragile this can be regarding paths, it might be better to just fork off a pkg-config run with `--atleast-version`

Reply:
(Debian/Ubuntu centric view) I was looking more closely at what `fwupdate`, `libfwup` and `libfwup-dev` provide.  Unfortunately `fwup.pc` is provided by `libfwup-dev` which is not installed by default when you install `fwupdate` from the repos or install Ubuntu (which has fwupdate preinstalled).

So I don't think this approach is going to work very well either.